### ISSUE TITLE
Improve performance by avoiding record updates

### ISSUE THREAD

**Issue by robinheghan** [folkertdev/elm-deque#2]
Hi!

I'm currently implementing my own `deque` and I use this library for comparison in tests and benchmarks. There's one ting that shows up in profiling and that is that record updates is where the majority of time is spent.

You can improve the performance of this library *significantly* if you replace all record updates with record statements.

So, if you have something like this:

```elm
type alias Something =
    { a : String
    , b : Int
    }
```

Then you should turn all things like this:

```elm
{ old | a = "new" }
```

Into

```elm
{ a = "new"
, b = old.b
}
```

**Comment by folkertdev** [comment]
Thanks! 

With almost every function updating a record the cost of the updates was probably large. 
It's a little unfortunate that we have to write slightly non-intuitive code to get good performance, but it's an easy change. 

I've also taken the time to update the tests to 0.19 and polish the docs a little. All of this was released as version 3.0.1.

**Comment by robinheghan** [comment]
Cool.

Based on my benchmark suite, it seems performance doubled with v3.0.1. :D
